src/ol/render/canvas/Executor.js

```diff
@@ -503,7 +524,7 @@ class Executor {
    */
   replayImageOrLabel_(
     context,
-    contextScale,
+    scaledCanvasSize,
```

Comment:
This change is to avoid having to access the context's canvas when recording instructions on the `ZIndexContextProxy`, because that canvas is not available.